From 43db095db210cb2b016d8301b15a66ae06a8a0b5 Mon Sep 17 00:00:00 2001
From: Sam Thursfield <sam.thursfield@codethink.co.uk>
Date: Tue, 5 May 2026 12:53:16 +0100
Subject: [PATCH 08/19] FDSDK 26.08: Use ModemManager from FDSDK

ModemManager was not part of FDSDK 25.08, but it is available in the
upcoming 26.08 release.
---
 elements/core-deps/ModemManager.bst     | 28 -------------------------
 elements/core-deps/NetworkManager.bst   |  2 +-
 elements/core-deps/fwupd.bst            |  2 +-
 elements/core/calls.bst                 |  2 +-
 elements/core/gnome-control-center.bst  |  2 +-
 elements/core/gnome-settings-daemon.bst |  2 +-
 elements/gnomeos-deps/81voltd.bst       |  2 +-
 7 files changed, 6 insertions(+), 34 deletions(-)
 delete mode 100644 elements/core-deps/ModemManager.bst

diff --git a/elements/core-deps/ModemManager.bst b/elements/core-deps/ModemManager.bst
deleted file mode 100644
index c1ff6e88d..000000000
--- a/elements/core-deps/ModemManager.bst
+++ /dev/null
@@ -1,28 +0,0 @@
-kind: meson
-
-sources:
-- kind: git_repo
-  url: freedesktop:mobile-broadband/ModemManager.git
-  track: '*.*'
-  exclude:
-  - '*-dev'
-  ref: 1.24.2-0-gf2b9ab1ad78d322f32134a444b5b54c6e8160e19
-
-build-depends:
-- sdk/gobject-introspection.bst
-- sdk/gtk-doc.bst
-- freedesktop-sdk.bst:public-stacks/buildsystem-meson.bst
-- freedesktop-sdk.bst:components/systemd.bst
-
-depends:
-- core-deps/libmbim.bst
-- core-deps/libqmi.bst
-- sdk/glib.bst
-- freedesktop-sdk.bst:components/dbus.bst
-- freedesktop-sdk.bst:components/libgudev.bst
-- freedesktop-sdk.bst:components/polkit.bst
-- freedesktop-sdk.bst:public-stacks/runtime-minimal.bst
-
-variables:
-  conf-local: >-
-    -Dudevdir=/usr/lib/udev
diff --git a/elements/core-deps/NetworkManager.bst b/elements/core-deps/NetworkManager.bst
index 914922d40..f5f2a4837 100644
--- a/elements/core-deps/NetworkManager.bst
+++ b/elements/core-deps/NetworkManager.bst
@@ -18,7 +18,6 @@ build-depends:
 - freedesktop-sdk.bst:public-stacks/buildsystem-meson.bst
 
 depends:
-- core-deps/ModemManager.bst
 - core-deps/mobile-broadband-provider-info.bst
 - core-deps/ppp.bst
 - sdk/glib.bst
@@ -31,6 +30,7 @@ depends:
 - freedesktop-sdk.bst:components/libndp.bst
 - freedesktop-sdk.bst:components/libnvme.bst
 - freedesktop-sdk.bst:components/libpsl.bst
+- freedesktop-sdk.bst:components/modemmanager.bst
 - freedesktop-sdk.bst:components/nss.bst
 - freedesktop-sdk.bst:components/polkit.bst
 - freedesktop-sdk.bst:components/systemd-libs.bst
diff --git a/elements/core-deps/fwupd.bst b/elements/core-deps/fwupd.bst
index 816c6e4e0..8c5629b90 100644
--- a/elements/core-deps/fwupd.bst
+++ b/elements/core-deps/fwupd.bst
@@ -20,7 +20,6 @@ build-depends:
 - freedesktop-sdk.bst:public-stacks/buildsystem-meson.bst
 
 depends:
-- core-deps/ModemManager.bst
 - core-deps/passim.bst
 - sdk/gcab.bst
 - sdk/glib.bst
@@ -39,6 +38,7 @@ depends:
 - freedesktop-sdk.bst:components/libsmbios-maybe.bst
 - freedesktop-sdk.bst:components/libusb.bst
 - freedesktop-sdk.bst:components/libxmlb.bst
+- freedesktop-sdk.bst:components/modemmanager.bst
 - freedesktop-sdk.bst:components/polkit.bst
 - freedesktop-sdk.bst:components/pycairo.bst
 - freedesktop-sdk.bst:components/python3-pil.bst
diff --git a/elements/core/calls.bst b/elements/core/calls.bst
index 3284ae492..a7e26fb37 100644
--- a/elements/core/calls.bst
+++ b/elements/core/calls.bst
@@ -18,7 +18,6 @@ build-depends:
 - freedesktop-sdk.bst:public-stacks/buildsystem-meson.bst
 
 depends:
-- core-deps/ModemManager.bst
 - core-deps/callaudiod.bst
 - core-deps/evolution-data-server.bst
 - core-deps/feedbackd.bst
@@ -31,4 +30,5 @@ depends:
 - sdk/gtk.bst
 - sdk/libadwaita.bst
 - freedesktop-sdk.bst:components/gstreamer.bst
+- freedesktop-sdk.bst:components/modemmanager.bst
 - freedesktop-sdk.bst:public-stacks/runtime-minimal.bst
diff --git a/elements/core/gnome-control-center.bst b/elements/core/gnome-control-center.bst
index 63894ea6d..fe0ac6ef1 100644
--- a/elements/core/gnome-control-center.bst
+++ b/elements/core/gnome-control-center.bst
@@ -29,7 +29,6 @@ runtime-depends:
 - core-deps/power-profiles-daemon.bst
 
 depends:
-- core-deps/ModemManager.bst
 - core-deps/NetworkManager.bst
 - core-deps/accountsservice.bst
 - core-deps/colord-gtk.bst
@@ -58,6 +57,7 @@ depends:
 - sdk/libadwaita.bst
 - freedesktop-sdk.bst:components/gstreamer.bst
 - freedesktop-sdk.bst:components/libwacom.bst
+- freedesktop-sdk.bst:components/modemmanager.bst
 - freedesktop-sdk.bst:components/pwquality.bst
 - freedesktop-sdk.bst:public-stacks/runtime-minimal.bst
 
diff --git a/elements/core/gnome-settings-daemon.bst b/elements/core/gnome-settings-daemon.bst
index b805e66cf..c4aa7d047 100644
--- a/elements/core/gnome-settings-daemon.bst
+++ b/elements/core/gnome-settings-daemon.bst
@@ -16,7 +16,6 @@ build-depends:
 - freedesktop-sdk.bst:public-stacks/buildsystem-meson.bst
 
 depends:
-- core-deps/ModemManager.bst
 - core-deps/NetworkManager.bst
 - core-deps/colord.bst
 - core-deps/cups-pk-helper.bst
@@ -35,6 +34,7 @@ depends:
 - freedesktop-sdk.bst:components/geoclue-libs.bst
 - freedesktop-sdk.bst:components/libgudev.bst
 - freedesktop-sdk.bst:components/libwacom.bst
+- freedesktop-sdk.bst:components/modemmanager.bst
 - freedesktop-sdk.bst:components/systemd-libs.bst
 - freedesktop-sdk.bst:components/xorg-lib-xtst.bst
 - freedesktop-sdk.bst:public-stacks/runtime-minimal.bst
diff --git a/elements/gnomeos-deps/81voltd.bst b/elements/gnomeos-deps/81voltd.bst
index 3a17807e7..bfa36eec6 100644
--- a/elements/gnomeos-deps/81voltd.bst
+++ b/elements/gnomeos-deps/81voltd.bst
@@ -12,7 +12,7 @@ build-depends:
 - freedesktop-sdk.bst:public-stacks/buildsystem-meson.bst
 
 depends:
-- core-deps/ModemManager.bst
+- freedesktop-sdk.bst:components/modemmanager.bst
 - gnomeos-deps/qrtr.bst
 
 config:
-- 
2.54.0

